### ISSUE TITLE
Functionality documentation: add documentation

### ISSUE THREAD

**Issue by rreinecke** [gdede-hua/Decision_Survey_Platform#4]
https://github.com/openjournals/joss-reviews/issues/4403

You nicely describe the use of the software in your paper draft but apart from the readme I was not able to find any proper documentation of the internal software components.

- Please document the API and provide guidance on where which component can be found e.g. you apply a MVC pattern to implement you tool. That should be at least pointed out.
- Provide guidance on how users can modify the software and where contributors may start when they 1) want to add new things to the user interface or 2) modify the internal algorithm
- So far the documentation of the source code is not acceptable. Just picking one file by random: https://github.com/gdede-hua/Decision_Survey_Platform/blob/master/app-maven/src/main/java/com/hua/model/FactorAnswer.java There are no docstrings that tell me what the class FactorAnswer is suppose to do. Please add extensive documentation that allows others to understand and reuse your code: https://www.oracle.com/de/technical-resources/articles/java/javadoc-tool.html

**Comment by fabian-s** [comment]
Seconded. 
See also #6

**Comment by gdede-hua** [comment]
Dear @rreinecke thank you for your comments.
We have included the code information in the readme.md (https://github.com/gdede-hua/Decision_Survey_Platform#code-informations).
Moreover a contributing.md file has been created in order to provide the guidance (https://github.com/gdede-hua/Decision_Survey_Platform/blob/master/Docs/Contributing.md).
Finally, detailed comments have been added in the actual code.

**Comment by rreinecke** [comment]
Great that you have included that information!

Concerning the code I could not find any substantial improvements of the code documentation on the main branch. Am I on the wrong branch?

For example https://github.com/gdede-hua/Decision_Survey_Platform/blob/master/app-maven/src/main/java/com/hua/util/RunAHP.java to my understanding implements the AGP algorithm and nothing is documented.

**Comment by fabian-s** [comment]
I tend to agree with @rreinecke here, not a lot of very useful documentation is present/has been added in the current master since April.
JOSS requires at least a complete API method documentation, which doesn't seem to exist here yet.

@gdede-hua: please also note that these reviews become a lot simpler for us reviewers/editors if your commit messages explicitly reference the respective issues they are meant to fix, or if your messages in these issue threads explicitly reference the commits that were intended to fix them.

**Comment by gdede-hua** [comment]
Dears, may I kindly ask you to provide me more information about what you expect in the API method documentation?

**Comment by fabian-s** [comment]
@gdede-hua: well, precise and correct descriptions of the classes you define and their methods need to be accessible in some way, at least for the user-facing ones. AFAICT, your code remains mostly undocumented.

@rreinecke could you provide some more specific technical advice on how to do this the right way? 
I don't program in Java myself.

**Comment by rreinecke** [comment]
I think this blog provides a nice overview: https://www.jrebel.com/blog/tips-and-tricks-for-better-java-documentation

**Comment by fabian-s** [comment]
@gdede-hua any news on this?

**Comment by gdede-hua** [comment]
@fabian-s please be kindly informed that I will address all the comments by end of January. Thank you.

**Comment by skatz91** [comment]
Dear @fabian-s & @rreinecke 
[commit](https://github.com/gdede-hua/Decision_Survey_Platform/commit/c8b0d6e3afb42301068fdafa7ddaff75542ab3c4)

I have added a lot of information for our function. Could you let me know if you want more changes?

Best regards
John Nikolaou

**Comment by fabian-s** [comment]
These are not sufficiently useful or informative comments for  documenting an API in my opinion-- see the very first comment by @rreinecke in this issue on the standard of documentation expected for JOSS-published software.
